Replicator/IncomingRev.cc

```diff
@@ -119,13 +120,73 @@ namespace litecore { namespace repl {
         if (_revMessage->noReply())
             _revMessage = nullptr;
 
-        _mayContainBlobs = jsonBody.containsBytes("\"digest\""_sl);
+        auto checkBlob = [](bool isDelta, alloc_slice jsonBody) -> bool {
+            if (!isDelta){
+                return jsonBody.containsBytes("\"digest\""_sl);
+            }
+
+            alloc_slice fleeceBody = impl::JSONConverter::convertJSON(jsonBody);
+            Value v = FLValue_FromData((C4Slice)fleeceBody, kFLTrusted);
+            Dict dictBody = v.asDict();
+            if (!dictBody) {
+                // It should be a dictionary. But in case, simply fallback to the old way.
+                return jsonBody.containsBytes("\"digest\""_sl);
+            }
+
+            for (DeepIterator iter(dictBody); iter; ++iter) {
+                if (iter.key() == C4Blob::kLegacyAttachmentsProperty) {
+                    //_attachments
+
+                    if (Array arr = iter.value().asArray(); arr) {
+                        // _attachments: [] or [ new value ]
+                        // JSON diff syntax for overwrite or delete
+                        return true;
+                    }
+                    if (Dict attachments = iter.value().asDict(); attachments) {
+                        for (Dict::iterator attIter(attachments); attIter; ++attIter) {
+                            if (Dict att = attIter.value().asDict(); att) {
+                                if (att.get(C4Blob::kDigestProperty)) {
+                                    return true;
+                                }
+                            } else if (Array arr = attIter.value().asArray(); arr) {
+                                // _attachments: { blob_/attached/1: [] or [ new value ] }
+                                // JSON diff syntax for overwrite or delete
+                                return true;
+                            }
+                        }
+                    }
+                    // We already inspected _attachments.
+                    iter.skipChildren();
+                } else {
+                    // Other than _attachments
+
+                    if (Dict dict = iter.value().asDict(); dict) {
+                        if (C4Blob::isBlob(dict) && dict.get(C4Blob::kDigestProperty)) {
+                            return true;
+                        }
+                    } else if (Array arr = iter.value().asArray(); arr && arr.count() == 1) {
```

Comment:
Do we need to look into the dictionary in the array here? Won't the `DeepIterator` iterate into the dictionaries in the array ?

Reply:
Here I am looking for new blob that is added. The JSON diff syntax is
property: [ newValue ]
the bracket is an artificial marker that indicates the value inside is to replace the original value. Example would be
`{"_attachments":{"blob_/attached/0":[],"blob_/attached/1":[],"blob_/attached/2":[]},"attached":{"3-":[{"@type":"blob","content_type":"text/plain","digest":"sha1-3jmj7l5rSw0yVb/vlWAYkK/YBwk=","length":10}]}} }`
"attached" is an array, but we use key, "3-", meaning from index 3 to the end of the array, replaced by a new blob
The new blob is denoted in square bracket, [], namely, arr[0] in the Value

Reply:
But I really don't need to do it here. Instead, I can, as you suggested, let the iterator run to the blob object and check it then.
Changed as suggested.